API/cpp/Src/DigitalOut.cpp

```diff
@@ -2,7 +2,7 @@
 
 DigitalOut::DigitalOut(PinName pin, PullType pull, PinMode mode, PinSpeed speed,
         bool inverted) : pin(pin), inverted(inverted) {
-    GPIO_InitTypeDef pin_structure;
+    GPIO_InitTypeDef pin_structure = {};
```

Comment:
Why this change, just curious?

Reply:
That was the issue we were having with SPI so I want to play it safe and zero out all structs like this